### PR TITLE
fix: favorite page image loading error, use skeleton animation

### PR DESCRIPTION
**Pull request by min-minnn**
### issue
- 로딩 이미지 안 어울림

### solution
- skeleton UI 사용해서 로딩되는 동안 애니메이션 적용
스크린샷 2024-05-27 오전 11 16 09

### CODE REVIEW

File: src/components/FavoritePage.vue [hunk 1]
```diff
@@ -16,16 +16,18 @@
...
       </div>
-      <div :class="{ 'click-disabled': this.isDisabled() == true }" class="w-full h-full mt-2 flex justify-center relative min-w-min">
+      <div :class="{ 'click-disabled': isDisabled() }" class="w-full h-full mt-2 flex justify-center relative min-w-min">
         <div @click="selectedImg(0)" :class="{ 'selected-left': this.$store.state.favoriteSelectedImg === 0 }" class="flex flex-col justify-center items-end cursor-pointer">
           <div :class="{ 'selected-left': this.$store.state.favoriteSelectedImg === 0, 'unselected-left': this.$store.state.favoriteSelectedImg === 1 }" class="w-favorite-content-width h-favorite-content-height aspect-w-1 aspect-h-1 border-8 cursor-pointer sm:min-w-72 min-h-72 overflow-hidden">
-            <img @error="handleImageError($event)" class="object-cover w-full h-full" :src="currentPair.image1" />
+            <div v-if="!imageLoaded1" class="skeleton-loader"></div>
+            <img v-show="imageLoaded1" @load="handleImageLoad(1)" @error="handleImageError(1)" class="object-cover w-full h-full" :src="currentPair.image1" />
```
Comment (kim-song-jun) [inline src/components/FavoritePage.vue]: use Array or Object for Loaded variable

File: src/components/FavoritePage.vue [hunk 2]
```diff
@@ -16,16 +16,18 @@
...
           </div>
           <div :class="{ 'selected-left-text': this.$store.state.favoriteSelectedImg === 0, 'hidden': this.$store.state.favoriteSelectedImg === 1 }" class="flex items-center justify-center w-favorite-content-width text-white text-2xl mt-3 min-w-72 sm:text-2xl selected-text-outline">{{ currentPair.title1 }}</div>
         </div>
         <div @click="selectedImg(1)" :class="{ 'selected-right': this.$store.state.favoriteSelectedImg === 1 }" class="flex flex-col justify-center cursor-pointer">
           <div :class="{ 'selected-right': this.$store.state.favoriteSelectedImg === 1, 'unselected-right': this.$store.state.favoriteSelectedImg === 0 }" class="w-favorite-content-width h-favorite-content-height aspect-w-1 aspect-h-1 border-8 cursor-pointer sm:min-w-72 min-h-72 overflow-hidden">
-            <img @error="handleImageError($event)" class="object-cover w-full h-full" :src="currentPair.image2" />
+            <div v-if="!imageLoaded2" class="skeleton-loader"></div>
+            <img v-show="imageLoaded2" @load="handleImageLoad(2)" @error="handleImageError(2)" class="object-cover w-full h-full" :src="currentPair.image2" />
```
Comment (kim-song-jun) [inline src/components/FavoritePage.vue]: Do not use magic number
use object or enum type

File: src/store.js [hunk 3]
```diff
@@ -109,9 +111,10 @@ export default createStore({
       state.active = i
     },
     async getFavoriteData(context) {
-      const res = await axios.get(
-        `${context.state.favoriteHost}/favorite/start/user/${context.state.userId}`
-      )
+      const res = await axios({
+        'method': 'get',
```
Comment (kim-song-jun) [inline src/store.js]: change to `axios.get()`

File: src/store.js [hunk 4]
```diff
@@ -82,6 +83,7 @@ export default createStore({
         state.favoriteCurrentPairIndex = 0
       }
       state.favoriteSelectedImg = ""
+      this.$store.state.favoriteImageError = false
```
Comment (kim-song-jun) [inline src/store.js]: Change to `state.favoriteImageError = false`

**Review by kim-song-jun — COMMENTED** [review]
👍

**Comment by min-minnn** [comment]
done

**Comment by min-minnn** [comment]
done

**Comment by min-minnn** [comment]
@kim-song-jun

File: src/components/FavoritePage.vue [hunk 5]
```diff
@@ -16,16 +16,18 @@
         </div>
         <div class="w-full text-white text-xl flex justify-center items-center pt-5">{{ this.$store.state.favoriteGameRound }}</div>
       </div>
-      <div :class="{ 'click-disabled': this.isDisabled() == true }" class="w-full h-full mt-2 flex justify-center relative min-w-min">
+      <div :class="{ 'click-disabled': isDisabled() }" class="w-full h-full mt-2 flex justify-center relative min-w-min">
         <div @click="selectedImg(0)" :class="{ 'selected-left': this.$store.state.favoriteSelectedImg === 0 }" class="flex flex-col justify-center items-end cursor-pointer">
           <div :class="{ 'selected-left': this.$store.state.favoriteSelectedImg === 0, 'unselected-left': this.$store.state.favoriteSelectedImg === 1 }" class="w-favorite-content-width h-favorite-content-height aspect-w-1 aspect-h-1 border-8 cursor-pointer sm:min-w-72 min-h-72 overflow-hidden">
-            <img @error="handleImageError($event)" class="object-cover w-full h-full" :src="currentPair.image1" />
+            <div v-if="!imagePair[0]" class="skeleton-loader"></div>
```
Comment (kim-song-jun) [inline src/components/FavoritePage.vue]: Do not use Magic number. using `enum` type or `object` type.

```js
const ImageIndex = {
  left: 0,
  right: 1
}


selectImage(ImageIndex.left)
!imagePair(ImageIndex.left)

```

File: src/store.js [hunk 6]
```diff
@@ -8,20 +8,21 @@ import game from "./data/gameData.js";
...
 
 // for Prod
+
 const TIMER = "https://api.timer.play-tino.com";
 const QUIZ = "https://api.quiz.play-tino.com";
 const FAVORITE = "https://api.favorite.play-tino.com";
 
 const FOOTERMENU = ["Contact", "Our Team", "Social"];
-const USERID = "3978099b-419d-46cb-a2ca-258b7f7ee535"; // for Test
+const USERID = "604bfbe0-0ecf-4d08-8c42-579c6f5da7a9"; // for Test
```
Comment (kim-song-jun) [inline src/store.js]: Change to `USER_ID`

File: src/store.js [hunk 7]
```diff
@@ -8,20 +8,21 @@ import game from "./data/gameData.js";
 // const FAVORITE = 'http://43.201.78.161:8080'
 
 // for Prod
+
 const TIMER = "https://api.timer.play-tino.com";
 const QUIZ = "https://api.quiz.play-tino.com";
 const FAVORITE = "https://api.favorite.play-tino.com";
 
 const FOOTERMENU = ["Contact", "Our Team", "Social"];
```
Comment (kim-song-jun) [inline src/store.js]: change to `FOOTER_MENU`

File: src/store.js [hunk 8]
```diff
@@ -342,7 +334,7 @@ export default createStore({
           })
         )
       })
-
+
```
Comment (kim-song-jun) [inline src/store.js]: check space

**Review by kim-song-jun — COMMENTED** [review]
👍 
Do not use Magic number and check variable name convention.

**Comment by min-minnn** [comment]
@kim-song-jun Done

**Comment by min-minnn** [comment]
@kim-song-jun Done

File: src/components/FavoritePage.vue [hunk 9]
```diff
@@ -39,14 +41,24 @@
...
+}
 
 export default {
   components: {
     FinishModal
   },
+  data() {
+    return {
+      imagePair: [false, false],
+      imageIndex: ImageIndex
```
Comment (kim-song-jun) [inline src/components/FavoritePage.vue:56]: use `Computed` when using enum type

**Review by kim-song-jun — COMMENTED** [review]
👍

**Comment by min-minnn** [comment]
@kim-song-jun

File: src/components/FavoritePage.vue (hunk 9 above)
Comment (kim-song-jun) [reply src/components/FavoritePage.vue:56]: Does it resolve?

**Review by kim-song-jun — COMMENTED** [review]
Check comment

File: src/components/FavoritePage.vue (hunk 9 above)
Comment (min-minnn) [reply src/components/FavoritePage.vue:56]: yes

Comment (kim-song-jun) [reply src/components/FavoritePage.vue:56]: but It's still in the data, and it seems didn't use computed

Comment (kim-song-jun) [reply src/components/FavoritePage.vue:56]: I will merge first, and fix it later

**Pull request merged**